src/app/api/events/[eventId]/volunteers/[volunteerId]/check-in/route.ts

```diff
@@ -1,7 +1,80 @@
-import { deleteEventVolunteer } from '@/server/actions/Volunteer';
+import {
+  deleteEventVolunteer,
+  updateVolunteer,
+} from '@/server/actions/Volunteer';
 import { zObjectId } from '@/types/dataModel/base';
 import { NextRequest, NextResponse } from 'next/server';
 import { mongo } from 'mongoose';
+import {
+  CreateEventVolunteerRequest,
+  zCreateEventVolunteerRequest,
+} from '@/types/dataModel/eventVolunteer';
+import { checkInVolunteer } from '@/server/actions/Event';
+import { UpdateVolunteerRequest } from '@/types/dataModel/volunteer';
+
+// @route POST /api/events/[eventId]/volunteers/[volunteerId]/check-in - Create an EventVolunteer
+export async function POST(
+  request: NextRequest,
+  { params }: { params: { eventId: string; volunteerId: string } }
+) {
+  try {
+    // Check Volunteer ID
+    const validationVolunteer = zObjectId.safeParse(params.volunteerId);
+    if (!validationVolunteer.success) {
+      return NextResponse.json(
+        { message: 'Invalid Volunteer Id' },
+        { status: 400 }
+      );
+    }
+
+    // Check Event ID
+    const validationEvent = zObjectId.safeParse(params.eventId);
+    if (!validationEvent.success) {
+      return NextResponse.json(
```

Comment:
Are we using CMerror?

Reply:
That'll be part of Dillon's PR -- not too concerned about errors in this one